src/models/model.cpp

```diff
@@ -455,8 +455,15 @@ void Model::CreateSessionOptionsFromConfig(const Config::SessionOptions& config_
       }
 
       session_options.AppendExecutionProvider("QNN", opts);
-    } else
+    } else if (provider_options.name == "WebGPU") {
```

Comment:
nit: 'webgpu' would be consistent with how other provider names are defined in the genai config currently